pkg/api/db.go

```diff
@@ -44,6 +44,23 @@ func CreateVolumeDBEntry(ctx *c.Context, in *model.VolumeSpec) (*model.VolumeSpe
 		log.Error(errMsg)
 		return nil, errors.New(errMsg)
 	}
+        if in.SnapshotId != "" {
+                snap, err := db.C.GetVolumeSnapshot(ctx, in.SnapshotId)
+                if err != nil {
+                        log.Error("Get snapshot failed in create volume method: ", err)
+                        return nil, err
+                }
+                if snap.Status != model.VolumeSnapAvailable {
+                        var errMsg = "Only the snapshot is available, the volume can be created"
```

Comment:
s/Only the snapshot/Only if the snapshot

Reply:
Done